editor/src/core/shared/jsx-attributes.ts

```diff
@@ -859,6 +861,17 @@ function walkAttribute(
   }
 }
 
+export function getAccumulatedElementsWithin(attributes: JSXAttributes): ElementsWithin {
+  let elementsWithinAccumulator: ElementsWithin = {}
+  walkAttributes(attributes, (attribute, path) => {
+    if (isJSXAttributeOtherJavaScript(attribute)) {
+      Object.assign(elementsWithinAccumulator, attribute.elementsWithin)
```

Comment:
Is there a reason for the `Object.assign` here, rather than just `elementsWithinAccumulator = attribute.elementsWithin`?

Reply:
this code is copying the values from attribute.elementsWithin to elementsWithinAccumulator

Usually we would write `elementsWithinAccumulator = {...elementsWithinAccumulator, ... attribute.elementsWithin}` but that desugars to `Object.assign({}, elementsWithinAccumulator, attribute.elementsWithin)` which copies elementsWithinAccumulator _and_  attribute.elementsWithin  as well into a new empty object.

While writing it I got nervous about not blowing the performance on a hot path such as the renderCoreElement